### PR TITLE
tree2: Add schema linting configuration

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Sometimes we need to make weird schema for testing or compatibility purposes. This allows us to still inform people when they do it on accident, but opt out of such checks as needed. This will also make adding new checks in the future more acceptable.

Also removes unneeded SchemaCollection abstraction layer.

## Breaking Changes

Schema builder constructor now has an extra argument that must be provided (default to {}) before the list of libraries if providing libraries.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: experimental/dds/tree2/src/feature-libraries/typed-schema/buildViewSchemaCollection.ts [hunk 1]
```diff
@@ -96,7 +103,7 @@ export function buildViewSchemaCollection(
 	};
 }
 
-export interface ViewSchemaCollection2 {
+export interface ViewSchemaCollection {
```
Comment (jenn-le) [inline experimental/dds/tree2/src/feature-libraries/typed-schema/buildViewSchemaCollection.ts:106]: Doc please

File: experimental/dds/tree2/src/feature-libraries/typed-schema/schemaBuilder.ts [hunk 2]
```diff
@@ -56,7 +58,12 @@ export class SchemaBuilder {
 	 * @param name - Name used to refer to this builder in error messages. Has no impact on the actual generated schema.
 	 * @param libraries - Libraries to include in this one. See `addLibraries` for details.
 	 */
-	public constructor(public readonly name: string, ...libraries: SchemaLibrary[]) {
+	public constructor(
+		public readonly name: string,
+		lint: Partial<SchemaLintConfiguration> = {},
```
Comment (jenn-le) [inline experimental/dds/tree2/src/feature-libraries/typed-schema/schemaBuilder.ts:63]: This needs to be added to the param docs.

File: experimental/dds/tree2/src/feature-libraries/untypedTree.ts [hunk 3]
```diff
@@ -105,8 +106,8 @@ export interface UntypedTreeCore<TContext = UntypedTreeContext, TField = Untyped
 	 * The type of the node.
 	 * If this node is well-formed, it must follow this schema.
 	 */
-	// TODO: update implementation to ensure a NamedTreeSchema is returned, and view schema is used in typed views.
-	readonly [typeSymbol]: NamedTreeSchema;
+	// TODO: update implementation to use view schema is used in typed views.
```
Comment (jenn-le) [inline experimental/dds/tree2/src/feature-libraries/untypedTree.ts:109]:
```suggestion
	// TODO: update implementation to use view schema in typed views.
```

File: experimental/dds/tree2/src/test/shared-tree/sharedTree.bench.ts [hunk 4]
```diff
@@ -38,14 +38,14 @@ const nodesCountDeep = [
 	[100, BenchmarkType.Measurement],
 ];
 
-const deepBuilder = new SchemaBuilder("sharedTree.bench: deep", jsonSchema);
+const deepBuilder = new SchemaBuilder("sharedTree.bench: deep", {}, jsonSchema);
```
Comment (jenn-le) [inline experimental/dds/tree2/src/test/shared-tree/sharedTree.bench.ts:41]: Having to pass this in when you have libraries isn't great. Maybe a config object would be better?

**Review by jenn-le — APPROVED** [review]

File: experimental/dds/tree2/src/test/shared-tree/sharedTree.bench.ts (hunk 4 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/test/shared-tree/sharedTree.bench.ts:41]: I'm not convinced `const deepBuilder = new SchemaBuilder("sharedTree.bench: deep", {libraries: [jsonSchema]});` is better, but maybe? I don't really like any of the options and I did consider that one. Something to get rethought in API review later I guess.

File: experimental/dds/tree2/src/feature-libraries/typed-schema/buildViewSchemaCollection.ts (hunk 1 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/feature-libraries/typed-schema/buildViewSchemaCollection.ts:106]:
```suggestion
/**
 * View Schema information collected from a SchemaBuilder.
 * Internal type for unifying document schema and libraries for use in validation and conversion.
 */
export interface ViewSchemaCollection {
```

**Pull request merged**